Begin replacing WebKitTestRunner injected bundle use with a UI-process alternative that works with site isolation

achristensen07 opened the pull request:
#### 1929f1302c0ef91bd357a3df45e76927065874a0
<pre>
Begin replacing WebKitTestRunner injected bundle use with a UI-process alternative that works with site isolation
<a href="https://bugs.webkit.org/show_bug.cgi?id=297156">https://bugs.webkit.org/show_bug.cgi?id=297156</a>
<a href="https://rdar.apple.com/157892436">rdar://157892436</a>

Reviewed by Ryosuke Niwa.

Instead of using cacheTestRunnerCallback and callTestRunnerCallback to cache state in the injected bundle,
use a WKUserScript to set up the function to post a message to the UI process, keep the state in the UI process
including which frame and how to call the callback, then use it when we get a callback.

* LayoutTests/http/tests/site-isolation/mouse-events/iframes-with-tooltips.html:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/Shared/API/c/WKBase.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/APIScriptMessage.cpp:
(API::ScriptMessage::ScriptMessage):
(API::ScriptMessage::page const):
* Source/WebKit/UIProcess/API/APIScriptMessage.h:
* Source/WebKit/UIProcess/API/C/WKAPICast.h:
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.cpp:
(WKFrameInfoGetPage):
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(CompletionListener::create):
(CompletionListener::complete):
(CompletionListener::CompletionListener):
(WKCompletionListenerComplete):
(WKPageSetFullScreenClientForTesting):
(WKPageSetPageUIClient):
(WKPageEvaluateJavaScriptInMainFrame):
(WKPageEvaluateJavaScriptInFrame):
* Source/WebKit/UIProcess/API/C/WKPage.h:
* Source/WebKit/UIProcess/API/C/WKPageFullScreenClient.h:
* Source/WebKit/UIProcess/API/C/WKScriptMessageRef.cpp: Copied from Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h.
(WKScriptMessageGetTypeID):
(WKScriptMessageGetBody):
(WKScriptMessageGetFrameInfo):
* Source/WebKit/UIProcess/API/C/WKScriptMessageRef.h: Copied from Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h.
* Source/WebKit/UIProcess/API/C/WKUserContentControllerRef.cpp:
(WebScriptMessageClient::WebScriptMessageClient):
(WKUserContentControllerAddScriptMessageHandler):
* Source/WebKit/UIProcess/API/C/WKUserContentControllerRef.h:
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.cpp:
(WKUserScriptCreate): Deleted.
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUserContentController.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::runJavaScriptInMainFrame): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::installTooltipDidChangeCallback): Deleted.
(WTR::TestRunner::callTooltipDidChangeCallback): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::requestPointerLock):
(WTR::TestController::tooltipDidChange):
(WTR::TestController::willEnterFullScreen):
(WTR::TestController::beganExitFullScreen):
(WTR::TestController::listenForTooltipChanges):
(WTR::TestController::resetStateToConsistentValues):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::tooltipDidChange): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::tooltipRegistrationCallback):
(WTR::TestController::platformCreateWebView):

Canonical link: <a href="https://commits.webkit.org/298479@main">https://commits.webkit.org/298479@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25bcb0381cd99cc6bb3ade63bf92c4937cbb3640

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115631 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35294 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25817 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121678 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66142 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0dace732-b6dc-443e-b5f6-d4d4c6064133) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117520 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35975 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43877 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87837 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42463 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5ced2ddc-a03f-486e-8d5c-0134675f6484) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118579 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28698 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103773 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68237 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27835 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21889 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65356 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98075 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22010 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124827 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42526 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31881 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96590 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42893 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99963 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96378 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41632 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19492 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38397 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18483 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42417 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47989 "Built successfully") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41890 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45221 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43598 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -1801,6 +1812,28 @@ void TestController::configureViewForTest(const TestInvocation& test)
+    WKUserContentControllerAddUserScript(controller.get(), script.get());
+
+    // FIXME: Generalize this to be able to be used for different test callbacks.
+    WKUserContentControllerAddScriptMessageHandler(controller.get(), toWK("wktr").get(), [] (WKScriptMessageRef message, WKCompletionListenerRef, const void *) {
```
Not sure about the abbreviation of `wktr`. Maybe we should just spell it out: `webkitTestRunner`

achristensen07 committed:
Begin replacing WebKitTestRunner injected bundle use with a UI-process alternative that works with site isolation
https://bugs.webkit.org/show_bug.cgi?id=297156
rdar://157892436

Reviewed by Ryosuke Niwa.

Instead of using cacheTestRunnerCallback and callTestRunnerCallback to cache state in the injected bundle,
use a WKUserScript to set up the function to post a message to the UI process, keep the state in the UI process
including which frame and how to call the callback, then use it when we get a callback.

* LayoutTests/http/tests/site-isolation/mouse-events/iframes-with-tooltips.html:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/Shared/API/c/WKBase.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/APIScriptMessage.cpp:
(API::ScriptMessage::ScriptMessage):
(API::ScriptMessage::page const):
* Source/WebKit/UIProcess/API/APIScriptMessage.h:
* Source/WebKit/UIProcess/API/C/WKAPICast.h:
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.cpp:
(WKFrameInfoGetPage):
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(CompletionListener::create):
(CompletionListener::complete):
(CompletionListener::CompletionListener):
(WKCompletionListenerComplete):
(WKPageSetFullScreenClientForTesting):
(WKPageSetPageUIClient):
(WKPageEvaluateJavaScriptInMainFrame):
(WKPageEvaluateJavaScriptInFrame):
* Source/WebKit/UIProcess/API/C/WKPage.h:
* Source/WebKit/UIProcess/API/C/WKPageFullScreenClient.h:
* Source/WebKit/UIProcess/API/C/WKScriptMessageRef.cpp: Copied from Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h.
(WKScriptMessageGetTypeID):
(WKScriptMessageGetBody):
(WKScriptMessageGetFrameInfo):
* Source/WebKit/UIProcess/API/C/WKScriptMessageRef.h: Copied from Source/WebKit/UIProcess/API/C/WKFrameInfoRef.h.
* Source/WebKit/UIProcess/API/C/WKUserContentControllerRef.cpp:
(WebScriptMessageClient::WebScriptMessageClient):
(WKUserContentControllerAddScriptMessageHandler):
* Source/WebKit/UIProcess/API/C/WKUserContentControllerRef.h:
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.cpp:
(WKUserScriptCreate): Deleted.
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUserContentController.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::runJavaScriptInMainFrame): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::installTooltipDidChangeCallback): Deleted.
(WTR::TestRunner::callTooltipDidChangeCallback): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::requestPointerLock):
(WTR::TestController::tooltipDidChange):
(WTR::TestController::willEnterFullScreen):
(WTR::TestController::beganExitFullScreen):
(WTR::TestController::listenForTooltipChanges):
(WTR::TestController::resetStateToConsistentValues):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::tooltipDidChange): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::tooltipRegistrationCallback):
(WTR::TestController::platformCreateWebView):

Canonical link: https://commits.webkit.org/298479@main
